files/common/usr/share/update-notifier/notify-reboot-required

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+# we do not include ". gettext.sh" here because:
+# a) it breaks if its not available
+# b) the string we have here does not need it (because it has no vars)
+eval_gettext() {
+    if [ -x /usr/bin/gettext ]; then
+        echo $(gettext "$1")
+    else
+        echo "$1"
+    fi
+}
+export TEXTDOMAIN=update-notifier
+export TEXTDOMAINDIR=/usr/share/locale
+
+case "$DPKG_MAINTSCRIPT_PACKAGE::$DPKG_MAINTSCRIPT_NAME" in
+    linux-image-extra*::postrm)
+        exit 0;;
+esac
+
+if [ "$0" = "/etc/kernel/postinst.d/update-notifier" ]; then
+    DPKG_MAINTSCRIPT_PACKAGE=linux-base
+fi
```

Comment:
Will this code ever run? I know this PR is simply copying over the script from upstream as-is but I wonder what happens if future upstream changes could break this script.

Reply:
as is, probably not.. but with the `update-notifier-common` package installed, there's a `/etc/kernel/postinst.d/update-notifier` file that's a symlink to `/usr/share/update-notifier/notify-reboot-required`.. and I think the scripts in `/etc/kernel/postinst.d` get execute when the kernel is upgraded (which is how the `reboot-required` file is created when a new kernel is installed)..

in this PR so far, I'm not creating that symlink.. but maybe I should? I thought about creating it, to remain consistent, and probably let it work for kernel upgrades too (the ZFS postinst hook calls `/usr/share/update-notifier/notify-reboot-required` directly, so practically, it probably doesn't matter for our appliance since we update ZFS every release)..

Reply:
But should we be concerned about copying a script from a third party package instead of actually depending on the package? By copying the script, aren't we opening ourselves up to the possibility of an upstream change breaking the script? Such breakages wouldn't happen if we simply depended on the third party package.
I guess both options have trade-offs. Depending on the third-party package bloats our product, but not adding it adds risk of breakages. If I were to pick, the added bloat is more acceptable than the latter.

Reply:
> If I were to pick, the added bloat is more acceptable than the latter.

it's not just bloat.. if we pull in `update-notifier-common`, we'll also pull in `ubuntu-advantage-tools`.. and the `ubuntu-advantage-tools` package has cause bugs, e.g. ESCL-4493..

but even if we fix those bugs, such that it is just "bloat".. I'd argue implementing a solution that works and doesn't have such bloat, would be a better solution.. and implementing our own script create the `/var/run/reboot-required` file, and not having the bloat, is better..

but I'm open to other input here.. if other's also think depending on the package, instead of rolling our own script, and inheriting the bloat that comes with that, is a better solution.. I'd be OK with that.

I was assuming the bloat was not acceptable, since we've been trying to reduce that bloat; e.g. https://github.com/delphix/delphix-platform/pull/439 (the change that introduced this breakage).

> By copying the script, aren't we opening ourselves up to the possibility of an upstream change breaking the script?

generally, sure.. in this specific case, though, I'm not concerned..

for example, we're adding a script at a specific location on the root filesystem.. since we're not pulling in the package that would have also (tried to) put the script at that location, we don't have any conflicts, so I think we're safe to install our version of the script.. arguably, the `delphix-platform` package should now conflict with the `update-notifier-common` package, to help the package manager prevent both package from being installed..

so I think we're safe to install our own script at the same location..

now the contents of the script are currently being copy/pasted as you're pointing out.. but I don't see anything in the script that would break over time.. is there any specific parts of the script that you're concerned about? e.g. anything you think will break due to an upstream change?

with that said, I also think most of the script is useless for us, and really all we need is the last two lines.. i.e.
```
echo "*** $(eval_gettext "System restart required") ***" > /var/run/reboot-required
echo "$DPKG_MAINTSCRIPT_PACKAGE" >> /var/run/reboot-required.pkgs
```
so if you feel strongly, I could modify our version of the script to just write to these two files.. I think that's also accomplish our goals..

even further, I think we could modify the ZFS package `postinst` script to create the `/var/run/reboot-required` file directly, and not call this script.. since we always update the ZFS kernel modules in each release, I think this would work.. although, might not be the most elegant solution.. if we did this, we wouldn't even need to create this script..

Let me know what you think.. I see a few valid options:

1. depend on `update-notifier-common`, causes bloat (assume bugs are fixed)
2. copy/paste like I've done here, might break over time (I'm still doubtful, but breakage could be fixed)
3. don't copy/paste, rewrite script to be more simple; could this also break over time?
4. modify ZFS postinst script to create `reboot-required` directly (kind of awkward, as it's delphix specific)

IMO, if we're really concerned that copy/paste won't work moving forward, I think I'd opt for (3) rather than (1).. but I'm open to other opinions.

Reply:
>anything you think will break due to an upstream change?
I was mostly referring to any upstream change that changes the functionality or the command that this script depends on. Nothing specific.
I forgot about ESCL-4493. I personally like option 3 which at least reduces chances of breakages.

Reply:
> I was mostly referring to any upstream change that changes the functionality or the command that this script depends on.

sure.. I just don't see anything that the script depends on that'd break..

Reply:
> I personally like option 3 which at least reduces chances of breakages.

I pushed an update, let me know what you think.